libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -2,7 +2,27 @@
 
 #define AP_BATTMONITOR_SMBUS_PEC_POLYNOME 0x07 // Polynome for CRC generation
 
-#define BATTMONITOR_SMBUS_TEMP 0x08 // temperature register
+#define BATTMONITOR_SMBUS_TEMP 0x08   // temperature register
+#define BATTMONITOR_SMBUS_SERIAL 0x1C // serial number
+AP_BattMonitor_SMBus::AP_BattMonitor_SMBus(AP_BattMonitor &mon,
+                                           AP_BattMonitor::BattMonitor_State &mon_state,
+                                           AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+        : AP_BattMonitor_Backend(mon, mon_state),
+        _dev(std::move(dev))
+{
+    _mon._serial_numbers[_state.instance] = AP_BATT_SERIAL_NUMBER_DEFAULT;
+}
+
+/// read the battery_voltage and current, should be called at 10hz
+void AP_BattMonitor_SMBus::read(void)
+{
+    // nothing to be done here for actually interacting with the battery
+    // however we can use this to set any parameters that need to be set
+
+    if (_serial_number && _mon._serial_numbers[_state.instance] == -1) {
```

Comment:
Nitpicking: comparing *_serial_number* to 0 is nicer (since it isn't a boolean) and changing the order would probably lead to a faster evaluation of the condition on most of the times.